openmc/mgxs/library.py

```diff
@@ -280,15 +289,21 @@ def domains(self, domains):
 
             self._domains = domains
 
+    @energy_groups.setter
+    def energy_groups(self, energy_groups):
+        cv.check_type('energy groups', energy_groups, openmc.mgxs.EnergyGroups)
+        self._energy_groups = energy_groups
+
     @correction.setter
     def correction(self, correction):
         cv.check_value('correction', correction, ('P0', None))
         self._correction = correction
 
-    @energy_groups.setter
-    def energy_groups(self, energy_groups):
-        cv.check_type('energy groups', energy_groups, openmc.mgxs.EnergyGroups)
-        self._energy_groups = energy_groups
+    @legendre_order.setter
+    def legendre_order(self, legendre_order):
+        cv.check_type('legendre_order', legendre_order, Integral)
+        cv.check_greater_than('legendre_order', legendre_order, 0, equality=True)
+        self._legendre_order = legendre_order
```

Comment:
Where is the right place to check if `legendre_order is > 0`, then it doesn't make sense to have a `correction=="P0"`?  In the setter of each `legendre_order` and `correction`?

Of course, if you wanted to implement the extended transport approximation and allow the `0` in `P0` to be `legendre_order + 1`, then that would work too.